Resources/Computing/BatchSystems/Condor.py

```diff
@@ -78,25 +78,34 @@ def submitJob( self, **kwargs ):
     nJobs = kwargs.get( 'NJobs' )
     if not nJobs:
       nJobs = 1
+    numberOfProcessors = kwargs.get( 'NumberOfProcessors' )
+    wholeNode = kwargs.get( 'WholeNode' )
     outputDir = kwargs['OutputDir']
     executable = kwargs['Executable']
     submitOptions = kwargs['SubmitOptions']
 
+    if wholeNode:
+      requirements = '+RequiresWholeMachine=True\n Requirements = ( CAN_RUN_WHOLE_MACHINE ) && ( OpSys == "LINUX" )'
+    else:
+      requirements = 'Requirements = OpSys == "LINUX"'
```

Comment:
I don't get these changes, would you please add some comments?

Reply:
A whole-node job needs two items in Condor. Firstly, it needs to advertise itself as one that is a whole-machine job, so that the machine can identify the job as a whole-machine job for matchmaking.  "+RequiresWholeMachine=True" will do this. Secondly, the job needs to require that it be matched with a machine that runs whole machine jobs, which can be identified with "CAN_RUN_WHOLE_MACHINE".
More reference to https://htcondor-wiki.cs.wisc.edu/index.cgi/wiki?p=WholeMachineSlots

Reply:
OK thanks. Please just add this last comment in the code.